clang 22 false-positives -Wfunction-effects on openal-soft's constexpr
operator""_uz inside 'nonblocking' mixer functions; upstream appends
-Werror=function-effects after user CXXFLAGS so it cannot be overridden.
Keep the warning, drop the -Werror promotion.

--- a/CMakeLists.txt	2026-06-12 13:49:09.895496249 +0100
+++ b/CMakeLists.txt	2026-06-12 13:49:09.911976112 +0100
@@ -407,9 +407,6 @@
         list(APPEND C_FLAGS -Werror=undef -Werror=format-security
             $<$<COMPILE_LANGUAGE:C>:-Werror=implicit -Werror=incompatible-pointer-types
                 -Werror=int-conversion>)
-        if(HAVE_WFUNCTION_EFFECTS)
-            list(APPEND C_FLAGS $<$<COMPILE_LANGUAGE:CXX>:-Werror=function-effects>)
-        endif()
     endif()
 
     # NOTE: This essentially provides the equivalent of the C++26 feature to
